pkg/operator/status/management_state.go

```diff
@@ -0,0 +1,40 @@
+package status
+
+import (
+	"github.com/golang/glog"
+	"github.com/openshift/api/operator/v1"
+)
+
+var (
+	allowOperatorUnmanagedState = true
+)
+
+// SetOperatorAlwaysManaged is one time choice when an operator want to opt-out from supporting the "unmanaged" state.
+// This is a case of control plane operators or operators that are required to always run otherwise the cluster will
+// get into unstable state or critical components will stop working.
+func SetOperatorAlwaysManaged() {
+	allowOperatorUnmanagedState = false
+}
+
+func IsOperatorAlwaysManaged() bool {
+	return !allowOperatorUnmanagedState
+}
+
+// IsOperatorManaged indicates whether the operator management state allows the control loop to proceed and manage
+// the operand.
+func IsOperatorManaged(state v1.ManagementState) bool {
+	switch {
+	case state == v1.Managed || IsOperatorAlwaysManaged():
+		return true
+	case state == v1.Removed:
+		// If state is "removed" it means operator is still managing operand uninstall.
+		// Some operators (control plane) ignore this state.
+		return true
+	case state == v1.Unmanaged:
+		return false
+	}
+
+	// In case users set wrong value, we default to managed.
+	glog.Warningf("Unknown %q value used for operator management state", state)
```

Comment:
A real control loop will allow you to set the `ManagementStateFailing` condition consistently and it will ripple up to clusteroperator status.

Reply:
ok, convinced, I turned this to a small controller loop observing the management state.